extension/chrome/elements/compose-modules/compose-draft-module.ts

```diff
@@ -111,23 +121,43 @@ export class ComposeDraftModule extends ViewModule<ComposeView> {
           sendable.body['text/plain'] = `[cryptup:link:draft_compose:${this.view.draftId}]\n\n${sendable.body['text/plain'] || ''}`;
         }
         const mimeMsg = await sendable.toMime();
-        if (!this.view.draftId) {
-          const { id } = await this.view.emailProvider.draftCreate(mimeMsg, this.view.threadId);
+        if (!this.view.draftId || this.isLocalDraftId(this.view.draftId)) {
+          let draftId: string;
+          try {
+            ({ id: draftId } = await this.view.emailProvider.draftCreate(mimeMsg, this.view.threadId));
+            if (this.isLocalDraftId(this.view.draftId)) { // delete local draft if there is one
+              await storageLocalRemove([this.view.draftId]);
+            }
+          } catch (e) {
+            if (ApiErr.isNetErr(e)) {
+              draftId = await this.localDraftCreate(mimeMsg, this.view.threadId, msgData.recipients.to || [], msgData.subject);
```

Comment:
could add `this.view.S.cached('send_btn_note').text('Draft saved locally (offline)');`

Reply:
Done 👍